arrow-libs/fx/arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/RaceNTest.kt

```diff
@@ -211,29 +215,31 @@ class RaceNTest : ArrowFxSpec(
 
     "Cancelling race 3 cancels all participants" {
       checkAll(Arb.int(), Arb.int(), Arb.int()) { a, b, c ->
-        val s = Semaphore(0L)
```

Comment:
Does ths mean we will deprecate Semaphore for Channel?

Reply:
No, we deprecate `Semaphore` for `Semaphore` and `ConcurrentVar` for `Channel`.

However, `Channel` of type `RENDEZ_VOUS` (or channel size = 0) can work in a similar way as a `Semaphore` with `0` permits. Whilst a `Semaphore` with `0` permits is not supported by the KotlinX Semaphore.

So this is use-case specific but since is quite low-level concurrency code to test other concurrent operators we have a bit of a special case here.